R/Spectra-functions.R

```diff
@@ -73,18 +76,39 @@ addProcessing <- function(object, FUN, ...) {
 #'
 #' Clean spectrum by removing 0-intensity peaks.
 #'
-#' @inheritParams .remove_peaks
+#' @inheritParams .peaks_remove
 #'
 #' @return `matrix` with columns `"mz"` and `"intensity"`.
 #'
 #' @noRd
-.clean_peaks <- function(x, spectrumMsLevel, all = FALSE,
+.peaks_clean <- function(x, spectrumMsLevel, all = FALSE,
                          msLevel = spectrumMsLevel, ...) {
     if (!spectrumMsLevel %in% msLevel || !length(x))
         return(x)
     x[utils.clean(x[, "intensity"], all), , drop = FALSE]
 }
 
+#' @description
+#'
+#' Bin spectrum. Code taken from MSnbase:::bin_Spectrum
+#'
+#' @inheritParams .peaks_remove
+#'
+#' @return `matrix` with columns `"mz"` and `"intensity"`
+#'
+#' @noRd
+.peaks_bin <- function(x, spectrumMsLevel, binSize = 1L,
+                       breaks = seq(floor(min(x[, 1])),
+                                    ceiling(max(x[, 1])),
+                                    by = binSize), fun = sum,
+                       msLevel = spectrumMsLevel, ...) {
+    if (!(spectrumMsLevel %in% msLevel))
+        return(x)
+    bins <- .bin_values(x[, 2], x[, 1], binSize = binSize, breaks = breaks,
+                        fun = fun)
+    cbind(mz = bins$mids, intensity = bins$x)
+}
+
 #' @description
 #'
 #' Wrapper to allow use of the `isCentroided` function with the `.peaksapply`
```

Comment:
Shouldn't we just add `...` to the arguments of `.isCentroided` instead of creating this wrapper here? (if you like you could also put the body of `.isCentroided` into `.is_centroided_peaks` or `.peaks_is_centroided` and remove it from `functions-utils.R`).

Reply:
👍 moved the code to a `.peaks_is_centroided` function.